core/src/main/java/com/amplifyframework/core/model/AuthRule.java

```diff
@@ -57,74 +66,111 @@ public AuthRule(com.amplifyframework.core.model.annotations.AuthRule authRule) {
-        return Empty.check(this.identityClaim) || "cognito:username".equals(this.identityClaim) ?
-                "username" :
-                this.identityClaim;
+        final String cliGeneratedIdentityClaim = "cognito:username";
+        return Empty.check(identityClaim) || cliGeneratedIdentityClaim.equals(identityClaim)
+                ? DEFAULT_IDENTITY_CLAIM
+                : identityClaim;
     }
 
     /**
-     * Used to specify a custom claim.   Defaults to "cognito:groups" when using AuthStrategy.GROUPS.
+     * Used for dynamic group authorization.
+     * Defaults to "groups" when using AuthStrategy.GROUPS.
      *
-     * @return group claim
+     * @return name of a {@link ModelField} of type String or array of Strings which specifies a group or list of groups
+     * which should have access.
      */
-    public String getGroupClaim() {
-        return this.groupClaim;
+    @NonNull
+    public String getGroupsFieldOrDefault() {
+        return Empty.check(groupsField)
+                ? DEFAULT_GROUPS_FIELD
+                : groupsField;
     }
 
     /**
-     * Used for static group authorization.
+     * Used to specify a custom claim.
+     * Defaults to "cognito:groups" when using AuthStrategy.GROUPS.
      *
-     * @return array of groups which should have access
+     * @return group claim
      */
-    public List<String> getGroups() {
-        return this.groups;
+    @NonNull
+    public String getGroupClaimOrDefault() {
+        return Empty.check(groupClaim)
+                ? DEFAULT_GROUP_CLAIM
+                : groupClaim;
     }
 
     /**
-     * Used for dynamic group authorization.  Defaults to "groups" when using AuthStrategy.GROUPS.
+     * Used for static group authorization.
      *
-     * @return name of a {@link ModelField} of type String or array of Strings which specifies a group or list of groups
-     * which should have access.
+     * @return array of groups which should have access
      */
-    public String getGroupsFieldOrDefault() {
-        return Empty.check(this.groupsField) ? "groups" : this.groupsField;
+    @NonNull
+    public List<String> getGroups() {
+        return Immutable.of(groups);
     }
 
     /**
      * Specifies which {@link ModelOperation}s are protected by this {@link AuthRule}.  Any operations not included in
      * the list are not protected by default.
      * @return list of {@link ModelOperation}s for which this {@link AuthRule} should apply.
      */
+    @NonNull
     public List<ModelOperation> getOperationsOrDefault() {
-        if (Empty.check(this.operations)) {
-            return Arrays.asList(
-                    ModelOperation.CREATE,
-                    ModelOperation.UPDATE,
-                    ModelOperation.DELETE,
-                    ModelOperation.READ);
-        }
-        return this.operations;
+        return Immutable.of(Empty.check(operations)
+                ? Arrays.asList(ModelOperation.values())
```

Comment:
While this changes makes the code more concise, I think it's a little more readable to explicitly enumerate the 4 values that make up the default.   That also ensures that if any values are added to the `ModelOperation` enum for some reason, this set of default operations won't be changed unexpectedly.

Reply:
I think this change addresses your second concern better in that it ensures that every operation is included even if `ModelOperation` is modified in the future. However, I do agree that it was more readable initially, so I'll revert it back :)